### ISSUE TITLE
panic when inspecting `docker://alexl/fedoratest:deltas`

### ISSUE THREAD

**Issue by vrothberg** [containers/skopeo#901]
While reading through https://github.com/containers/image/pull/902, I wanted to have a look at the delta images and found Skopeo to panic:
```
~ $ skopeo inspect docker://alexl/fedoratest:deltas                          
panic: runtime error: invalid memory address or nil pointer dereference      
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x5642d5992d07]
...
```

`--raw` does not panic.

**Comment by alexlarsson** [comment]
I saw this at some point, fix is:
```
diff --cc manifest/oci_index.go
index 0553c73,0553c73..7bdea8f
--- a/manifest/oci_index.go
+++ b/manifest/oci_index.go
@@@ -79,6 -79,6 +79,9 @@@ func (index *OCI1Index) ChooseInstance(
        }
        for _, wantedPlatform := range wantedPlatforms {
                for _, d := range index.Manifests {
++                      if d.Platform == nil {
++                              continue
++                      }
                        imagePlatform := imgspecv1.Platform{
                                Architecture: d.Platform.Architecture,
                                OS:           d.Platform.OS,
```

**Comment by alexlarsson** [comment]
Otoh, maybe we should generate some fake os/arch as these are nominally required fields.

**Comment by vrothberg** [comment]
Nice. I didn't read the full stack to not get side tracked. As you have the fix at hand, would you open a PR against c/image?

The image-spec states that the platform is optional but that it "SHOULD be present if its target is platform-specific." As delta images are specific to image of certain platforms, I we should set the platform there as well.

**Comment by alexlarsson** [comment]
@vrothberg IT depends on your definition of "specific to". I mean you can apply deltas to any tarball, whether the result is target specific depends on the tarball itself. 

I guess we could copy the arch from the original layer info. Is that what you meant?

**Comment by vrothberg** [comment]
Yes, using the platform of the original image seems right.

**Comment by mtrmac** [comment]
This was brought in by #904 and released in Skopeo 1.0.0. Thanks again!
